src/erlang_ls_provider.erl

```diff
@@ -2,69 +2,64 @@
 
 -behaviour(gen_server).
 
+%% API
+-export([ initialize/1
+        , handle_request/2
+        , start_link/2
+        , teardown/2
+        ]).
+
+%% gen_server callbacks
+-export([ init/1
+        , handle_call/3
+        , handle_cast/2
+        ]).
+
 -callback is_enabled()    -> boolean().
 -callback setup(config()) -> state().
 -callback teardown()      -> ok.
 
--export([ enabled_providers/0
-        , is_enabled/1
```

Comment:
I removed from the export list most of these functions and had `initialize/1` use them internally instead.